### PR TITLE
Handle revoked oauth permissions by the user

### PR DESCRIPTION
**Pull request by humitos**
If the user revoked our app permission from the external service web page, we are not notified. So, next time we try to use it, we will receive a 401 status code in the first request.

Here we handle this case and we raised an Exception from the Celery task, which is shown in the UI while pulling the Public Task information.

Missing things:

- [x] ~implement a proper persistent message that the user can close~
- [ ] ~find a way to make a general test request from `create_session` instead of `paginate`~

Related: https://github.com/readthedocs/readthedocs-corporate/issues/323
Related: https://github.com/rtfd/readthedocs.org/pull/4070
~Based on: #4078~ (already merged)

### CODE REVIEW

**Comment by ericholscher** [comment]
This looks good, but I think @agjohnson is probably the one who should review it, since I'm not as familiar with this code.

File: readthedocs/oauth/services/base.py [hunk 1]
```diff
@@ -131,6 +138,26 @@ def paginate(self, url, **kwargs):
...
+
+            # TODO: this check of the status_code would be better in the
+            # ``create_session`` method since it could be used from outside, but
+            # I didn't find a generic way to make a test request to each
+            # provider.
+            if resp.status_code == 401:
+                # Bad credentials: the token we have in our database is not
+                # valid. Probably the user has revoked the access to our App. He
+                # needs to reconnect his account
+                message = 'Our access to your {} account was revoked. You ' \
```
Comment (humitos) [inline readthedocs/oauth/services/base.py]: Consdider to use a better general message here and also maybe catch the 404 --in a another PR with more testing, probably.

**Comment by humitos** [comment]
This is very close to be functional.

I used our custom notification app and the `Notification` class which uses the `SiteBackend` and it worked.

captura de pantalla_2018-06-04_17-54-09

There are still some problems: 
1. there is no way to close/accept/dismiss the notification once it's promped to the user
1. the notification is not immediately shown since it doesn't comes from the PublicTask --we need to refresh the page
1. once the notification is shown, if I go to the Admin settings, when I come back it disappears (it's marked as `read`)

**Comment by humitos** [comment]
Now, after implementing this and re read #4078 I'm thinking that we may want just to raise an Exception inside the task when we receive the 401 which should be catched by that PR and properly shown immediately after sync and not persistent.

**Comment by humitos** [comment]
We decided to not use persistent message for this.

So, I just based this branch in #4078 and raise a simple Exception with this message:

captura de pantalla_2018-06-04_18-28-57

**Comment by humitos** [comment]
This is ready to be merged after rebased and re-tested (QA).

**Pull request merged**